avocado/plugins/list.py

```diff
@@ -76,26 +76,13 @@ def _get_test_matrix(self, test_suite):
                 elif 'path' in params:
                     id_label = params['path']
 
-            try:
-                type_label = type_label_mapping[cls]
-                decorator = decorator_mapping[cls]
-                stats[type_label.lower()] += 1
-                type_label = decorator(type_label)
-            except KeyError:
-                if isinstance(cls, str):
-                    cls = test.Test
-                    type_label = type_label_mapping[cls]
-                    decorator = decorator_mapping[cls]
-                    stats[type_label.lower()] += 1
-                    type_label = decorator(type_label)
-                    id_label = params['name']
-                elif issubclass(cls, test.Test):
-                    cls = test.Test
-                    type_label = type_label_mapping[cls]
-                    decorator = decorator_mapping[cls]
-                    stats[type_label.lower()] += 1
-                    type_label = decorator(type_label)
-                    id_label = params['name']
+            if isinstance(cls, str):
+                cls = test.Test
+                id_label = params['name']
+            type_label = type_label_mapping[cls]
+            decorator = decorator_mapping[cls]
+            stats[type_label.lower()] += 1
+            type_label = decorator(type_label)
```

Comment:
I don't really remember what was the problem with inherited tests, but you are right, all instrumented tests should be discovered as `string`. Btw shouldn't we rather use `basestring` to include unicode names?